### DISCUSSION TITLE
ball / noballのビルドができません

### DISCUSSION THREAD

**Discussion by na260** [Yowkees/keyball#67] [Q&A] [answered]
先日は、センサーのご対応ありがとうございました。
非常に助かりました。
またしても質問させていただいて恐縮ですが、
最新ソースをcloneしてビルドしようとしましたが、以下の様なエラーでビルドできません。
お手数ですがご確認おねがいできますか？
(make keyball/rev1:via ですとビルドできるようです。)

```Compiling: keyboards/keyball/rev1/transport.c                                                      keyboards/keyball/rev1/transport.c:308:6: error: nested redefinition of ‘enum serial_transaction_id’
 enum serial_transaction_id {
      ^
keyboards/keyball/rev1/transport.c:308:6: error: redeclaration of ‘enum serial_transaction_id’
In file included from quantum/split_common/transactions.h:23:0,
                 from drivers/serial.h:22,
                 from keyboards/keyball/rev1/transport.c:240:
quantum/split_common/transaction_id_define.h:19:6: note: originally defined here
 enum serial_transaction_id {
```

下記の手順で実行しました
https://github.com/Yowkees/keyball#prerequirements-to-build-firmware

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。無事にセンサーが動作して良かったです。

Keyball46のファームウェアがmakeできない件について、QMK_Firmware 0.13.x までは正常でしたが、最新の QMK_Firmware 0.14.x にてmakeできないとのご連絡を頂いております。
この問題について昨日まで多忙だったため取り組めておりませんでしたが、今週時間を作れそうですので調査させていただきます。

申し訳ありませんが少々お時間をいただければ幸いです。

**Reply by na260** [reply]
ご返信ありがとうございます。
取り急ぎ0.13.x でmakeできることを確認しました。
私のほうは現状で不足はありませんので
ご多忙のところ恐縮ですが、お手すきでご対応お願いいたします。

**Reply by Yowkees** [reply]
お心遣いありがとうございます。
修正できましたらご連絡させていただきます

**Comment by Yowkees** [comment, marked as answer]
@na260 
最新のKeyball46のファームウェアにてQMK_Firmware 0.14.22でもmakeできるように修正しました。
KoRoN様の多大なるご協力により対応させることができました。大変感謝です。

さて、最新ファームウェアではQMK_Firmwareの機能増加に伴ってメモリ容量が増えたので、仕様が変わりました。
具体的にはVIA対応ファームウェアではOLEDの表示が出なくなります。その代わり選択できるLEDパターンが増えました。
ファームウェアをカスタマイズされるにはdefaultのキーマップを使用されるとよいと思います。

ご検討よろしくお願いします。
